source/generators/MuonGenerator.cc

```
@@ -66,105 +133,200 @@ MuonGenerator::MuonGenerator():
   }
 
   // Convert vector to arr
-  G4double arr_ang[v_angles.size()];
-  std::copy(v_angles.begin(), v_angles.end(), arr_ang);
+  auto arr_ang = v_angles.data();
 
   // Initialise the Random Number Generator based on cos(x)*cos(x) distribution
   fRandomGeneral_ = new G4RandGeneral( arr_ang, v_angles.size() );
 
 }
 
-
-MuonGenerator::~MuonGenerator()
+void MuonGenerator::SetupAngles()
 {
-
-  delete msg_;
+  // Rotation from the axes used in file.
+  // Rotates anticlockwise about Y.
+  rPhi_ = new G4RotationMatrix();
+  rPhi_->rotateY(-axis_rotation_);
 }
 
+
 void MuonGenerator::GeneratePrimaryVertex(G4Event* event)
 {
-  particle_definition_ = G4ParticleTable::GetParticleTable()->FindParticle(MuonCharge());
+
+  // Initalise RNG seeds and load file on the first event
+  if (!bInitialize_){
+
+    // Load in the Muon angular distribution from file
+    if (use_lsc_dist_){
+      std::cout << "[MuonGenerator]: Generating muons using lsc distribution loaded from file" << std::endl;
+      LoadMuonDistribution();
+    }
```

As a protection against a possible user's mistake of setting both a specific direction and `use_lsc_dist` to `true`, we could raise an error after line 161, `if user_dir_ != G4ThreeVector{}`.